### PR TITLE
Self update fix

### PR DESCRIPTION
**Pull request by feinstruktur**
I've updated the `self` sub-commands `compile` and `update` to fix an issue where `Process.arguments[0]` is different depending on whether the script is text executable or a binary (see https://github.com/qutheory/vapor-cli/issues/12#issuecomment-223583417)

I've replaced the hard-coded install to `/usr/local/bin` with a flexible from:to: method which is called from both `compile` and `update`. The idea idea is that regardless of where you install the vapor script initially (just a `chmod +x` and a `cp` to a location) these two commands will always attempt to replace the location they are in (`pathToSelf`) with the result of the operation (`update` or `compile`).

I've also made `update` run `compile` automatically - I think that's always what you want, isn't it?

So in summary, users should be able to:

- download from the URL
- chmod +x
- cp anywhere
- run compile/update to their heart's content no matter where the script is and it will compile or update in place at that location

I've tested all bits on OSX and Linux except for update, which I can't really test with this new version.

### CODE REVIEW

**Comment by tannernelson** [comment]
This is really smart:

```swift
return try? runWithOutput("which vapor").trim()
```

Waiting on Travis to pass and I'll include in 0.5.2
